go/vt/vtgate/executor.go

```diff
@@ -104,10 +104,10 @@ func NewExecutor(ctx context.Context, serv srvtopo.Server, cell, statsName strin
 	e.vm.watchSrvVSchema(ctx, cell)
 
 	executorOnce.Do(func() {
-		stats.Publish("QueryPlanCacheLength", stats.IntFunc(e.plans.Length))
-		stats.Publish("QueryPlanCacheSize", stats.IntFunc(e.plans.Size))
-		stats.Publish("QueryPlanCacheCapacity", stats.IntFunc(e.plans.Capacity))
-		stats.Publish("QueryPlanCacheEvictions", stats.IntFunc(e.plans.Evictions))
+		stats.NewGaugeFunc("QueryPlanCacheLength", "Query plan cache length", e.plans.Length)
+		stats.NewGaugeFunc("QueryPlanCacheSize", "Query plan cache size", e.plans.Size)
+		stats.NewGaugeFunc("QueryPlanCacheCapacity", "Query plan cache capacity", e.plans.Capacity)
+		stats.NewGaugeFunc("QueryPlanCacheEvictions", "Query plan cache evictions", e.plans.Evictions)
 		stats.Publish("QueryPlanCacheOldest", stats.StringFunc(func() string {
```

Comment:
Again this can be deferred to a future PR, but this seems like a case in which the earlier proposal to add a conversion API could be helpful -- that way we could export this as a gauge.